### PR TITLE
first iterations of (hopefully) streamlined functionality

### PR DESCRIPTION
**Pull request by kaifox**
.... more or less as discussed ...

I tried to also take care about the peculiarities with `Maybe<Void>` ... for this I introduced additional methods (isSuccessful(), ifSuccessful(...)), which can be used in such situations, as the hasValue() will be false and the ifValue() would never be called .... 

Further, I tried to remove some redpeating lines of code, by using the attempt() method more often ....

### CODE REVIEW

File: src/main/java/org/ossgang/commons/monads/Maybe.java [hunk 1]
```diff
@@ -105,31 +116,64 @@ public boolean hasException() {
         return exception != null;
     }
 
+    /**
+     * Returns true, if this Maybe is in a successful state, i.e. it does not contain an exception and thus has a valid
+     * value. For normal Maybes, the value will always be non-null. The only exception to this is a Maybe<Void> which in
+     * successful state has a value of {@code null}. This is equivalent to {@code !hasException()}
```
Comment (michi42) [inline src/main/java/org/ossgang/commons/monads/Maybe.java]: "This is equivalent to {@code !hasException()}" - no, it is not for Maybe<Void>. I would just remove that mentioning here.

File: src/main/java/org/ossgang/commons/monads/Maybe.java [hunk 2]
```diff
@@ -105,31 +116,64 @@ public boolean hasException() {
...
+     *
+     * @return true if the the value is availabl, false if not
```
Comment (michi42) [inline src/main/java/org/ossgang/commons/monads/Maybe.java]: type: "the the value" and "availabl"

File: src/main/java/org/ossgang/commons/monads/Maybe.java [hunk 3]
```diff
@@ -88,12 +92,19 @@ private Maybe(Throwable exception) {
...
+     * Returns true, if this Maybe is in a successful state, i.e. it does not contain an exception. Usually this means,
+     * the Maybe has a valid (non-null) value. So ususally this is equivalent to {@code #hasValue()}. The only exception
+     * to this is a "successful"
+     * Maybe<Void> which does not have a valid value (and thus the {@link #value} would throw even if the monad is in a
+     * successful state.
+     * <p>
+     * This is equivalent to {@code !hasException()}
      *
-     * @return the contained exception, or Optional.empty() if none occurred.
+     * @return true if the the value is availabl, false if not
```
Comment (michi42) [inline src/main/java/org/ossgang/commons/monads/Maybe.java]: typo "the the value" and "availabl"

File: src/main/java/org/ossgang/commons/monads/Maybe.java (hunk 1 above)
Comment (kaifox) [reply src/main/java/org/ossgang/commons/monads/Maybe.java]: Indeed ... was a nasty moving/copying error ... I changed.

File: src/main/java/org/ossgang/commons/monads/Maybe.java (hunk 2 above)
Comment (kaifox) [reply src/main/java/org/ossgang/commons/monads/Maybe.java]: done

**Pull request merged**